src/components/icon/js/iconService.js

```diff
@@ -422,8 +422,25 @@
     */
    function cacheIcon( id ) {
 
-     return function updateCache( icon ) {
-       iconCache[id] = isIcon(icon) ? icon : new Icon(icon, config[id]);
+     return function updateCache( _icon ) {
+       var icon = isIcon(_icon) ? _icon : new Icon(_icon, config[id]);
+
+       //clear id attributes to prevent aria issues
+       var elem = icon.element;
+       elem.removeAttribute('id');
+
+       var children = elem.childNodes;
+
+       for (var i=0; i<children.length;i++) {
+         var child = children[i];
+         if(child.nodeName==='path') {
```

Comment:
The code is just called once per icon type, unconnected to the number of times i use the icon in the application, and even in ie11 the performance is 10^5 ops per second, so this should be fine.